tests/kernel/workq/work_queue/src/main.c

```diff
@@ -351,48 +352,42 @@ static int test_delayed(void)
 	return check_results(NUM_TEST_ITEMS);
 }
 
-
-void main(void)
+void testing_workq(void)
 {
-	int status = TC_FAIL;
+	int old_prio = k_thread_priority_get(k_current_get());
 
-	if (test_sequence() != TC_PASS) {
-		goto end;
-	}
+	k_thread_priority_set(k_current_get(), 0);
```

Comment:
what about this test requires that we change the thread priority?

Reply:
Without this test_delayed_resubmit_thread fails. When there was only main thread it was working fine so I just put all the test case to run in thread priority 0.